### PR TITLE
Rework data structure to be reasonable

### PR DESCRIPTION
**Pull request by LemmaEOF**
This cleans up a lot of the data stuff and moves away from being an unnecessary ECS. Primary changes:
- Character class and resource registries are new just new SimpleRegistries, instead of being bloaty subclasses.
- `CharacterClass` and `CharacterResource` (formerly `ClassComponentType` and `ResourceBarComponentType`, respectively) are now interfaces instead of abstract classes
- `CharacterClassEntry` and `CharacterResourceEntry` (formerly `ClassComponent` and `ResourceBarComponent`, respectively) are now classes instead of interfaces
- All data is managed through the `CharacterData` class (formerly `PlayerDataHolder`), which takes a PlayerEntity as input and provides access to their `CharacterClasses` and `CharacterResources` objects, which are the core objects for storing and interacting with character data. More features of each are currently planned.
- Replace demo classes with instances of `SimpleCharacterClass` and `SimpleCharacterResource`, constructed and registered inline in the main `CottonRPG` (formerly `CottonRPGMod`) class.
- `Ticker` is not currently used because I don't trust it to keep existing safely between ticks (Minecraft can be finnicky with things not saved to NBT). Will be worked on.
- Indentation remains untouched.

### CODE REVIEW

File: src/main/java/io/github/cottonmc/cottonrpg/CottonRPG.java [hunk 1]
```diff
@@ -2,21 +2,27 @@
...
 import net.minecraft.server.command.CommandManager;
 import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.SimpleRegistry;
+
+import java.awt.*;
 
-public class CottonRPGMod implements ModInitializer {  
+public class CottonRPG implements ModInitializer {
```
Comment (handicraftsman) [inline src/main/java/io/github/cottonmc/cottonrpg/CottonRPG.java:23]: Weeee, a mod class rename!

Comment (LemmaEOF) [reply src/main/java/io/github/cottonmc/cottonrpg/CottonRPG.java:23]: It's to fit in with the standard of everything else in the Cotton org. I changed the fabric.mod.json to match.

File: src/main/java/io/github/cottonmc/cottonrpg/data/CharacterClassEntry.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+/**
+ * Flyweight for interacting with classes.
+ * A CharacterClassEntry stores the Identifier of the class, the player's level of the class, and how much XP in the class the player has.
+ * Class XP is separate from a player's XP, and the XP in other classes.
+ * However, you may want to implement experience on your classes, so feel free to access this then.
+ */
+public class CharacterClassEntry {
+  public Identifier id;
+  private int level = 0;
+  private int experience = 0;
```
Comment (handicraftsman) [inline src/main/java/io/github/cottonmc/cottonrpg/data/CharacterClassEntry.java:15]: What if a mod has character-global experience and wants to allow leveling any available class on levelup? I think it should be possible to create class experience resource and bind it to the class.

**Review by handicraftsman — APPROVED** [review]
Generally nice, although has a flaw regarding class levelup. Still pretty nice. I understand now why you were against such container wrspper as i did.

**Comment by handicraftsman** [comment]
I think its mergeable. Waiting for falkreon now.

File: src/main/java/io/github/cottonmc/cottonrpg/data/CharacterClassEntry.java (hunk 2 above)
Comment (LemmaEOF) [reply src/main/java/io/github/cottonmc/cottonrpg/data/CharacterClassEntry.java:15]: I'm in favor of class-bound resources, but I don't think they should be saved within the class themselves. This experience variable is entirely optional; this is basically copy-pasted from SkillCheck which doesn't use it at all. We might wanna add a method to CharacterClass to list associated resources.

Comment (handicraftsman) [reply src/main/java/io/github/cottonmc/cottonrpg/data/CharacterClassEntry.java:15]: Yeah, true. 

Regarding class bound resources: avatar mod has 4 main elements and some more secondary ones. Using them uses a shared resource - chi.

Comment (falkreon) [reply src/main/java/io/github/cottonmc/cottonrpg/data/CharacterClassEntry.java:15]: As a compromise, you could have an optional pointer to a Resource which is the experience bar for this class

Comment (handicraftsman) [reply src/main/java/io/github/cottonmc/cottonrpg/data/CharacterClassEntry.java:15]: Yeah, that would be nice.

**Pull request merged**